woocommerce/woocommerce
# Restore "No shipping options were found" message in cart-shipping template.

-   I have followed the [WooCommerce Contributing Guidelines](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md) and the [WordPress Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/).
-   I have checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change.
-   I have reviewed my code for [security best practices](https://developer.wordpress.org/apis/security/).
-   Following the above guidelines will result in quick merges and clear and detailed feedback when appropriate.



### Changes proposed in this Pull Request:

In `cart-shipping.php` there was a check to see if the `$available_methods` was truthy, i.e. did it contain any available methods. If so, it would show the shipping methods. If not, it would show the "No shipping options were found" message.

In https://github.com/woocommerce/woocommerce/pull/43237 the code was updated to check if the variable is set, and if it's an array. This is true for **empty** arrays, so the condition evaluated to true for empty arrays.

To return to the original intention of the check, I propose using `empty` which encompasses two checks in one, it encompasses `isset` AND `count( $available_methods ) > 0` thus restoring the previous functionality.

### Submission Review Guidelines:





Closes #44920 .
Closes #45243



### How to test the changes in this Pull Request:



Using the [WooCommerce Testing Instructions Guide](https://github.com/woocommerce/woocommerce/wiki/Writing-high-quality-testing-instructions), include your detailed testing instructions:

1. Set up shipping zones on your site, but ensure there's a country with no shipping zones covering it.
2. Enable the shipping calculator (WooCommerce -> Settings -> Shipping -> Shipping Settings -> Enable the shipping calculator on the cart page)
3. Add an item to your cart and go to the (shortcode) Cart page.
4. Using the shipping calculator, enter an address for a country with no shipping zones.
5. Expect to see the "No shipping options were found" message.
6. Change to a country that _is_ covered by shipping zones. Expect the error to be gone, and for the correct zones to show.
7. Proceed to the (shortcode) Checkout.
8. Ensure the shipping methods are visible.
9. Change country to one that is not covered by any zone.
10. Ensure the "No shipping options were found" message shows in the shipping area.



### Changelog entry




<details>

#### Significance




#### Type




#### Message 

#### Comment 

</details>
